modules/apps/transfer/ibc_module.go

```diff
@@ -229,6 +230,11 @@ func (im IBCModule) OnAcknowledgementPacket(
 		return sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-20 transfer packet data: %s", err.Error())
 	}
 
+	bz := types.ModuleCdc.MustMarshalJSON(&ack)
+	if !bytes.Equal(bz, acknowledgement) {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "acknowledgement did not marshal to expected bytes: %X ≠ %X", bz, acknowledgement)
```

Comment:
code is [29](https://github.com/cosmos/cosmos-sdk/blob/release/v0.46.x/types/errors/errors.go#L125) instead of [12](https://github.com/cosmos/ibc-go/blob/main/modules/core/errors/errors.go#L49)?